src/subsplit_dag.cpp

```diff
@@ -578,6 +608,264 @@ void SubsplitDAG::ReversePostorderIndexTraversal(
+      }
+    }
+  }
+  // If the parent is new, connect it to all of its children except for the given child
+  // node (don't reindex these edges).
+  if (parent_is_new) {
+    CreateAndInsertNode(parent_subsplit);
+    new_node_ids.push_back(GetDAGNodeId(parent_subsplit));
+    const auto &children_of_parent = GetChildIds(parent_subsplit);
+    for (const auto &[rotated, child_of_parent_ids] : children_of_parent) {
+      SafeInsert(parent_to_range_, PerhapsRotateSubsplit(parent_subsplit, rotated),
+                 {GPCSPCountWithFakeSubsplits(),
+                  GPCSPCountWithFakeSubsplits() + child_of_parent_ids.size()});
+      for (const size_t child_of_parent_id : child_of_parent_ids) {
+        if (child_of_parent_id != GetDAGNodeId(child_subsplit)) {
+          new_edge_idxes.push_back(GPCSPCountWithFakeSubsplits());
+          CreateAndInsertEdge(GetDAGNodeId(parent_subsplit), child_of_parent_id,
+                              rotated);
+        }
+      }
+    }
+  }
+  // Connect the given parent node to the given child node (don't reindex these edges if
+  // the parent is new). `prev_edge_count` is a marker conveying where we need to start
+  // reindexing edge idxes.
+  size_t prev_edge_count = GPCSPCountWithFakeSubsplits();
+  new_edge_idxes.push_back(GPCSPCountWithFakeSubsplits());
+  CreateAndInsertEdge(GetDAGNodeId(parent_subsplit), GetDAGNodeId(child_subsplit),
+                      child_subsplit.SubsplitIsRotatedChildOf(parent_subsplit));
+  if (parent_is_new) {
+    prev_edge_count = GPCSPCountWithFakeSubsplits();
+  }
+  // Connect the child to all of its parents except for the given parent node (reindex
+  // these edges).
+  if (child_is_new) {
+    const auto &parents_of_child = GetParentIds(child_subsplit);
+    for (const auto &[rotated, parents_of_child_ids] : parents_of_child) {
+      for (const size_t parent_of_child_id : parents_of_child_ids) {
+        if (parent_of_child_id != GetDAGNodeId(parent_subsplit)) {
+          new_edge_idxes.push_back(GPCSPCountWithFakeSubsplits());
+          CreateAndInsertEdge(parent_of_child_id, GetDAGNodeId(child_subsplit),
+                              rotated);
+        }
+      }
+    }
+  }
+  // Connect the parent to all of its parents (reindex these edges).
+  if (parent_is_new) {
+    const auto &parents_of_parent = GetParentIds(parent_subsplit);
+    for (const auto &[rotated, parent_of_parent_ids] : parents_of_parent) {
+      for (const size_t parent_of_parent_id : parent_of_parent_ids) {
+        new_edge_idxes.push_back(GPCSPCountWithFakeSubsplits());
+        CreateAndInsertEdge(parent_of_parent_id, GetDAGNodeId(parent_subsplit),
+                            rotated);
+      }
+    }
+  }
+  // Create reindexers.
+  node_reindexer = BuildNodeReindexer(prev_node_count);
+  edge_reindexer = BuildEdgeReindexer(prev_edge_count);
+  // Update new_node_ids and new_edge_idxes according to the reindexers.
+  for (size_t &id : new_node_ids) {
```

Comment:
Can't we replace this with a function call?

Reply:
I don't think we should. We could, but it wouldn't be a reindex function call. This is simply updating the couple node ids we just added to where they are now after the reindexing. We are not reindexing new_node_ids and new_edge_idxs. I'd be happy to chat with you about this.

Reply:
I understand what's going on, and that it's not an existing reindexer function call. I just suggest adding another function, which would allow us to factor out this (albeit simple) behavior and attach a useful name to the operation. This also gives an opportunity to add a test. 

And this implementation seems a little "tricky"-- we are simultaneously using id and changing it via reference. An application of std::transform would seem more canonical.